docs/rfcs/001-updateable-bundled-packages.md

```diff
@@ -0,0 +1,100 @@
+
+Since a large part of Atom's built-in functionality is provided by bundled packages, it makes sense to allow some of those packages to be updated independently of Atom's monthly release cycle so that users can receive new features and fixes whenever they become available.
+
+The primary use case for this improvement is enabling the GitHub package to ship improvements more frequently than Atom's release cycle since many of its improvements can be done without changes to Atom itself.  If this approach is proven to work well for the GitHub package, we might also consider using it to ship Teletype as a bundled Atom package.
+
+## Explanation
+
+> Explain the proposal as if it was already implemented in Atom and you were describing it to an Atom user. That generally means:
+> - Introducing new named concepts.
+> - Explaining the feature largely in terms of examples.
+> - Explaining any changes to existing workflows.
+
+Bundled packages are treated differently than community packages that you can install using `apm`:
+
+- You are not prompted to update them when new versions are released on `apm`
+- `apm` will warn you at the command line when you try to install or update a bundled package
+- If a user intentionally installs a bundled package from `apm` the [Dalek package](https://github.com/atom/dalek/) will show a warning in the "deprecations" view asking the user to remove the offending package
+
+Despite all this, if the user *does* install a bundled package using `apm`, it will be loaded into the editor and updated dutifully as releases occur.
+
+### Implementation Details
+
+Because the necessary infrastructure is already in place to enable updates to bundled packages using `apm`, the only work required is to provide a way for packages to opt in to this behavior and for `apm` to include those packages in its update checks if they haven't already been installed in the user's packages folder.
+
+Any bundled Atom package will be able to opt in to updates by adding `"updateable": true` to its `package.json` file.  This will cause `apm` to consider it as part of the list of packages it checks for updates.  If a community (non-bundled) package sets this field to `true` or `false` it will be ignored as it's only relevant to bundled packages.
+
+`apm` will be updated to include the list of bundled packages with `"updateable": true` set in their `package.json` so that the user will be notified of new package versions that support the engine version of their current Atom build.
+
+### User Experience Examples
+
+1. The user downloads Atom 1.28.0 from atom.io which includes GitHub package version 0.15.0.  After Atom 1.28.0 was released, a hotfix release was shipped for the GitHub package as 0.15.1.  When the user installs and starts Atom, they are prompted to install the update to the GitHub package.
```

Comment:
Does this example imply that GitHub package 0.15.1 is released _between_ A) the time that the user downloads Atom 1.28.0 and B) the time they install Atom 1.28.0? If so, I think this could be written more generally to describe any scenario where GitHub package 0.15.1 is shipped after Atom 1.28.0 is released:

> The user downloads and installs Atom 1.28.0 from atom.io which includes GitHub package version 0.15.0. After Atom 1.28.0 was released, a hotfix release was shipped for the GitHub package as 0.15.1. When the user starts Atom, they are prompted to install the update to the GitHub package.

What do you think?

Reply:
Looks like I worded this one poorly, here's the scenario:

1. Atom 1.28.0 is released
2. One week later GitHub package 0.15.1 is released
3. Two weeks later a user installs Atom 1.28.0
4. Upon starting Atom, user is prompted to install GitHub 0.15.1.

Updated wording to better reflect the scenario:

> The user downloads and installs Atom 1.28.0 which was released two weeks prior.  Since that release the GitHub package has been updated to version 0.15.1 on `apm`.  When the user starts Atom for the first time, they are prompted to update the GitHub package.